### PR TITLE
Add draft of a concepts page

### LINKED ISSUE #166: Write a `Concepts` page
Explain `Project`, `Check`, etc.

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #166.

### CODE REVIEW

File: website/src/docs/concepts.mdx [hunk 1]
```diff
@@ -5,3 +5,108 @@ export const info = {
...
+{/* TODO: Link to or inline a small example of this? */}
+
+You can run `Executable`s with `garn run`.
+
+## Packages
+
+`Package`s are instructons to `garn` about how to _build_ a set of files. For
+example for a go backend a package would define how to compile it into
+executables. For an npm project it may define how the project can be bundled
+into a set of files that can be served by a webserver.
```
Comment (soenkehahn) [inline website/src/docs/concepts.mdx:89]: Add a comment that `Package`s are built in sandboxes.

File: website/src/docs/concepts.mdx [hunk 2]
```diff
@@ -5,3 +5,108 @@ export const info = {
...
+- Running a code formatter,
+- Run a code generator,
+- Spin up a database for development,
+- etc.
+
+{/* TODO: Link to or inline a small example of this? */}
+
+You can run `Executable`s with `garn run`.
+
+## Packages
```
Comment (soenkehahn) [inline website/src/docs/concepts.mdx:84]: Add links to api docs for all the entities.

File: website/src/docs/concepts.mdx [hunk 3]
```diff
@@ -5,3 +5,108 @@ export const info = {
 };
 
 # Basic concepts
+
+There's 4 underlying entities that play a role in `garn`:
+
+- `Environment`s
+- `Check`s
+- `Executable`s
+- `Package`s
```
Comment (soenkehahn) [inline website/src/docs/concepts.mdx:14]: Add links to API docs.

File: website/src/docs/concepts.mdx [hunk 4]
```diff
@@ -5,3 +5,108 @@ export const info = {
...
+
+## Checks
+
+`Check`s are commands (usually shell commands) that are used to define
+reproducible automated tests or other checks. They are being run in a sandbox,
+which means they don't have access to your locally installed tools, nor to your
+local file system. Instead they are always based on an underlying `Environment`,
+and through that will have access to the tools that the environment provides. In
+addition they will have access to a _copy_ of your source files, that is
+provided in the sandbox they are run in. That gives you the ability to define
```
Comment (alexdavid) [inline website/src/docs/concepts.mdx:51]:
```suggestion
addition, a _copy_ of your source files are
provided in the sandbox. That gives you the ability to define
```

**Review by alexdavid — COMMENTED** [review]
Couple of small nits TIOLI

File: website/src/docs/concepts.mdx [hunk 5]
```diff
@@ -5,3 +5,108 @@ export const info = {
...
+- Spin up a database for development,
+- etc.
+
+{/* TODO: Link to or inline a small example of this? */}
+
+You can run `Executable`s with `garn run`.
+
+## Packages
+
+`Package`s are instructons to `garn` about how to _build_ a set of files. For
```
Comment (alexdavid) [inline website/src/docs/concepts.mdx:86]:
```suggestion
`Package`s are instructions to `garn` about how to _build_ a set of files. For
```

**Review by alexdavid — APPROVED** [review]
This looks great!

**Pull request merged**